src/lephare/data_manager.py

```diff
@@ -126,3 +126,35 @@ def create_work_subdirectories(self, parent_dir):
         work_sub_directories = ["filt", "lib_bin", "lib_mag", "zphota"]
         for sub_dir in work_sub_directories:
             os.makedirs(os.path.join(parent_dir, sub_dir), exist_ok=True)
+
+    def remove_empty_run_directories(self):
+        """Remove any empty run directories from the cache. If any file exists in
+        any of the subdirectories, the run directory is not considered empty.
+        Note that we generate a new run directory if the work directory is no longer
+        a symlink."""
+        default_os_cache = user_cache_dir("lephare", ensure_exists=True)
+        work_sub_directories = ["filt", "lib_bin", "lib_mag", "zphota"]
+        runs_dir = os.path.join(default_os_cache, "runs")
+        work_dir = os.path.join(default_os_cache, "work")
+        symlink_target = os.readlink(work_dir)
+
+        # Remove empty directories (only if all subdirs are empty)
+        for run_name in os.listdir(runs_dir):
+            # print(run_name)
+            run_dir = os.path.join(runs_dir, run_name)
+            empty = True
+            for sub_dir in work_sub_directories:
+                # print(os.listdir(os.path.join(run_dir, sub_dir)))
+                if len(os.listdir(os.path.join(run_dir, sub_dir))) > 0:
+                    empty = False
+                    break
+            if empty:
+                print(f"-> Removing empty run directory: {run_dir}")
+                for sub_dir in work_sub_directories:
```

Comment:
Initially I thought that there was a parameter that could be set in os.rmdir to recursively remove a directory, but it looks like in the python docs that it's actually a different function, `shutil.rmtree(run_dir)`. I'm not sure it's worth swapping it out in the interest of fewer lines of code or not.

Reply:
Yeah! I'd thought about using shutil, but liked avoiding the import and there's something about being explicit when deleting subdirs that helps calm the nervousness of messing with a user's files. Haha. Maybe I'm too shy about this, but I like that this will fail if there's some unexpected other subdirectory present here. I do agree the extra lines aren't as nice as a single "remove directory" line would be. And that's enough overthinking for today 🙃

Reply:
Cool, that all makes sense to me, and it a great justification for keeping it as is. Thanks.